zebra-chain/src/transaction.rs

```diff
@@ -1036,8 +1036,39 @@ impl Transaction {
 
     /// Access the note commitments in this transaction, if there are any,
     /// regardless of version.
-    pub fn orchard_note_commitments(&self) -> Box<dyn Iterator<Item = &pallas::Base> + '_> {
-        orchard_shielded_data_iter!(self, orchard::ShieldedData::note_commitments)
+    pub fn orchard_note_commitments(&self) -> Box<dyn Iterator<Item = pallas::Base> + '_> {
+        match self {
```

Comment:
It would be nice to use the `orchard_shielded_data_iter!` macro here too, it could be updated to accept a second argument for mapping v5 transactions:

```rust
macro_rules! orchard_shielded_data_iter {
    ($self:expr, $mapper:expr) => {
        orchard_shielded_data_iter!($self, $mapper, $mapper)
    };

    ($self:expr, $mapper:expr, $mapper_v5:expr) => {
        match $self {
            // No Orchard shielded data
            Transaction::V1 { .. }
            | Transaction::V2 { .. }
            | Transaction::V3 { .. }
            | Transaction::V4 { .. } => Box::new(std::iter::empty()),

            Transaction::V5 {
                orchard_shielded_data,
                ..
            } => Box::new(orchard_shielded_data.into_iter().flat_map($mapper)),

            #[cfg(feature = "tx-v6")]
            Transaction::V6 {
                orchard_shielded_data,
                ..
            } => Box::new(orchard_shielded_data.into_iter().flat_map($mapper_v5)),
        }
    };
}
```